apps/api/lib/api/user/user.ex

```diff
@@ -8,4 +8,8 @@ defmodule API.User do
   def create(attrs), do: DB.User.create(attrs)
 
   def login(attrs), do: Authorization.authorize(attrs)
+
+  def get_by_username(username), do: DB.User.get_by_username(username)
+  def search_users(username: username), do: DB.User.search_users(username: username)
```

Comment:
since we pattern match on the key value pair at the db layer, i dont think we need to do it at the api layer. we can probably consolidate search users into one function.